SwiftCompilerSources/Sources/Optimizer/Utilities/AccessUtils.swift

```diff
@@ -329,7 +329,13 @@ struct AccessPath : CustomStringConvertible {
     if !base.isEqual(to: other.base) {
       return nil
     }
-    return projectionPath.subtract(from: other.projectionPath)
+    if let resultPath = projectionPath.subtract(from: other.projectionPath),
+       // Indexing is not a projection where the base overlaps the "projected" address.
+       !resultPath.pop().kind.isIndexedElement
```

Comment:
Do you need to check if `resultPath` contains any indexedElement as opposed to only checking the back of the subpath? You can index into an array then project a field from the indexed element.

Reply:
`pop` returns the _first_ path component and not the last one. But it's a good point and I'll add test cases for both variants.

Reply:
> `pop` returns the _first_ path component and not the last one. But it's a good point and I added a test case for it

Oh I see. That totally makes sense.